heartbeat/tests/system/test_icmp.py

```diff
@@ -44,8 +45,12 @@ def has_failed_message(): return self.log_contains("Failed to initialize ICMP lo
         # we should run pings on those machines and make sure they work.
         self.wait_until(lambda: has_started_message() or has_failed_message(), 30)
 
+        self.wait_until(lambda: self.output_has(lines=1))
+        output = self.read_output()
+        monitor_status = output[0]["monitor.status"]
+        monitor_error = output[0]["error.message"]
         if has_failed_message():
-            proc.check_kill_and_wait(1)
+            assert monitor_status == "down"
+            self.assertRegex(monitor_error, ".*Insufficient privileges to perform ICMP ping.*")
         else:
-            # Check that documents are moving through
-            self.wait_until(lambda: self.output_has(lines=1))
+            assert monitor_status == "up"
```

Comment:
Yeah, I think it's a good point for a follow-up PR perhaps.

The only to determine if you can do a ping is to actually do one sadly or maybe to use `getcap`. 

The tricky thing from a CI perspective is you need to see if you have permission, somehow, to run `sudo setcap cap_net_raw+eip` on the heartbeat binary. I think we'd need to execute this from python as well.

We probably need this logic regardless because on most people's laptops the no pings behavior will be the default.

I'm not averse to us reaching out to robots to fix this, but I don't think it's a priority given the ultra-low churn WRT the ICMP code.